server/src/test/java/org/elasticsearch/transport/TransportHandshakerTests.java

```diff
@@ -184,10 +189,17 @@ public void testHandshakeResponseFromOlderNodeWithPatchedProtocol() {
                 containsString("which has an incompatible wire format")
             )
         );
+        assertDeprecationMessageIsLogged(remoteVersion, releaseVersion, channel);
+    }
+
+    private void assertDeprecationMessageIsLogged(TransportVersion remoteVersion, String remoteReleaseVersion, Object channel) {
+        if (remoteVersion.onOrAfter(TransportVersions.MINIMUM_COMPATIBLE) && remoteVersion.before(V8_18_FIRST_VERSION)) {
+            assertWarnings(getDeprecationMessage(TransportVersion.current(), remoteVersion, remoteReleaseVersion, channel));
+        }
     }
 
-    private static TransportVersion getRandomIncompatibleTransportVersion() {
-        return randomBoolean()
+    private static TransportVersion getRandomIncompatibleTransportVersion(boolean olderThanMinCompatible) {
```

Comment:
Is this new parameter needed? It looks like we just pass in a `randomBoolean()` and don't otherwise care?

Reply:
This is a leftover from a prior iteration, I'll remove it, thanks!